internal/consensus/ticker.go

```diff
@@ -29,17 +29,23 @@ type TimeoutTicker interface {
 type timeoutTicker struct {
 	service.BaseService
 
-	timer    *time.Timer
-	tickChan chan timeoutInfo // for scheduling timeouts
-	tockChan chan timeoutInfo // for notifying about them
+	timerActive bool
+	timer       *time.Timer
+	tickChan    chan timeoutInfo // for scheduling timeouts
+	tockChan    chan timeoutInfo // for notifying about them
+	stopChan    chan struct{}    // for stopping the timeout routine
 }
 
 // NewTimeoutTicker returns a new TimeoutTicker.
 func NewTimeoutTicker() TimeoutTicker {
 	tt := &timeoutTicker{
-		timer:    time.NewTimer(0),
-		tickChan: make(chan timeoutInfo, tickTockBufferSize),
-		tockChan: make(chan timeoutInfo, tickTockBufferSize),
+		timer: time.NewTimer(0),
+		// An indicator variable to check if the timer is active or not.
+		// Concurrency safe because the timer is only accessed by a single goroutine.
+		timerActive: true,
+		tickChan:    make(chan timeoutInfo, tickTockBufferSize),
+		tockChan:    make(chan timeoutInfo, tickTockBufferSize),
+		stopChan:    make(chan struct{}),
```

Comment:
don't think we need it. after `OnStop` is called, `t.Quit()` is closed.

Reply:
Thank you! Wasn't familiar with thats how stop works, I'll remove that